### PR TITLE
Add: advanced cron-like scheduling for queries

### PR DESCRIPTION
**Pull request by luke14free**
Same as https://github.com/getredash/redash/pull/1425 but on master, closing the old PR.
50ee033a-b16c-11e6-98d9-50b5e2d0e5e8

### CODE REVIEW

**Comment by luke14free** [comment]
can I do anything to facilitate the review/merge of this PR? :)

File: redash/models.py [hunk 1]
```diff
@@ -685,6 +688,9 @@ def should_schedule_next(previous_iteration, now, schedule):
             previous_iteration = normalized_previous_iteration - datetime.timedelta(days=1)
 
         next_iteration = (previous_iteration + datetime.timedelta(days=1)).replace(hour=hour, minute=minute)
+    else:
+        future_dates = croniter(schedule, previous_iteration)
+        next_iteration = future_dates.get_next(datetime.datetime)
```
Comment (arikfr) [inline redash/models.py]: This is too critical to go without tests.

**Comment by arikfr** [comment]
Please reopen when updated. Thanks!

**Comment by oytuntez** [comment]
this would help a lot! :)

[rq-scheduler](https://github.com/rq/rq-scheduler) now supports cron-like scheduling, [schedule.py](https://github.com/getredash/redash/blob/master/redash/tasks/schedule.py) can be extended to support it.

**Pull request closed without merging**